### PR TITLE
Support for testing sharded queues

### PR DESCRIPTION
**Pull request by dmcintyre-pivotal**
This PR adds support for testing sharded queues.

A new option -S/--slow-start inserts a 1 second delay between starting each consumer.

If a queue is predeclared (-p), then it will not be bound to the exchange.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@dmcintyre-pivotal Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/rabbitmq?repositoryId=rabbitmq/rabbitmq-perf-test&pullRequestId=57)!

[Click here](https://cla.pivotal.io/sync/rabbitmq?repositoryId=rabbitmq/rabbitmq-perf-test&pullRequestId=57) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@dmcintyre-pivotal Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/rabbitmq?repositoryId=rabbitmq/rabbitmq-perf-test&pullRequestId=57)!

File: src/main/java/com/rabbitmq/perf/MulticastParams.java [hunk 1]
```diff
@@ -294,7 +303,7 @@ public boolean shouldConfigureQueues() {
             generatedQueueNames.add(qName);
             // skipping binding to default exchange,
             // as it's not possible to explicitly bind to it.
-            if (!"".equals(exchangeName) && !"amq.default".equals(exchangeName)) {
+            if (!predeclared && !"".equals(exchangeName) && !"amq.default".equals(exchangeName)) {
```
Comment (michaelklishin) [inline src/main/java/com/rabbitmq/perf/MulticastParams.java]: Is this change really related?

File: src/main/java/com/rabbitmq/perf/MulticastSet.java [hunk 2]
```diff
@@ -116,6 +116,10 @@ public void run(boolean announceStartup)
 
         for (Thread consumerThread : consumerThreads) {
             consumerThread.start();
+            if(params.getConsumerSlowStart()) {
+            	System.out.println("slowStart: wait 1sec");
```
Comment (michaelklishin) [inline src/main/java/com/rabbitmq/perf/MulticastSet.java]: Can we make it say "Delaying start of next consumer for 1 second because of -S…"?

File: src/main/java/com/rabbitmq/perf/MulticastParams.java (hunk 1 above)
Comment (dmcintyre-pivotal) [reply src/main/java/com/rabbitmq/perf/MulticastParams.java]: Yes, because if you try to bind a sharded queue to an exchange you get an error.

Plus, if a queue is predeclared, allowing the perf-test tool to add another binding to it is probably not what you want.

Comment (michaelklishin) [reply src/main/java/com/rabbitmq/perf/MulticastParams.java]: That makes sense but will break some scenarios, see c061038a62f6e0583d0ef4aaab6293bb8f4cdd97 and #43.

Comment (michaelklishin) [reply src/main/java/com/rabbitmq/perf/MulticastParams.java]: @dmcintyre-pivotal do you think we should introduce a new boolean flag? I'm afraid this and #43 expect two completely different behaviours. `--skip-binding-queues`, perhaps?

**Comment by michaelklishin** [comment]
@acogoluegnes I will. PerfTest doc guide could use an expansion in general.
